File: Data Structures & Algorithms/insert-new-interval/submission-1.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        int a,b;
        a=newInterval[0];b=newInterval[1];
        int n=intervals.size();vector<vector<int>> res;int i=0;
while(i<n && intervals[i][1]<a){
    res.push_back(intervals[i]);i++;
}
while(i<n && intervals[i][0]<=b){
    a=min(a,intervals[i][0]);//key step
    b=max(b,intervals[i][1]);//key step
    i++;
}
res.push_back({a,b});
while(i<n){
res.push_back(intervals[i]);i++;
}
return res;
    }
};

File: Data Structures & Algorithms/merge-intervals/submission-1.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda i:i[0]) #sort wrt start time
        curr=intervals[0]
        res=[]
        for i in range(1,len(intervals)):
            if curr[1]<intervals[i][0]:
            # add curr to res
                res.append(curr.copy())
                curr=intervals[i]
            else:
                curr=[curr[0],max(curr[1],intervals[i][1])]
        res.append(curr)
        return res

        